util/drakeGeometryUtil.cpp

```diff
@@ -135,6 +135,55 @@ double quatNorm(const Eigen::Vector4d& q)
   return std::acos(q(0));
 }
 
+Eigen::Vector4d slerp(const Eigen::Vector4d& q1, const Eigen::Vector4d& q2, double interpolation_parameter)
+{
+  /* Q = slerp(q1, q2, f) Spherical linear interpolation between two quaternions
+  *   This function uses the implementation given in Algorithm 8 of [1].
+  *
+  * @param q1   Initial quaternion (w, x, y, z)
+  * @param q2   Final quaternion (w, x, y, z)
+  * @param f    Interpolation parameter between 0 and 1 (inclusive)
+  * @retval Q   Interpolated quaternion(s). 4-by-1 vector.
+  *
+  * [1] Kuffner, J.J., "Effective sampling and distance metrics for 3D rigid
+  * body path planning," Robotics and Automation, 2004. Proceedings. ICRA '04.
+  * 2004 IEEE International Conference on , vol.4, no., pp.3993,3998 Vol.4,
+  * April 26-May 1, 2004
+  * doi: 10.1109/ROBOT.2004.1308895
+  */
+
+  // Compute the quaternion inner product
+  double lambda = (q1.transpose() * q2).value();
+  int q2_sign;
+  if (lambda < 0.0) {
+    // The quaternions are pointing in opposite directions, so use the equivalent alternative representation for q2
+    lambda = -lambda;
+    q2_sign = -1;
+  }
+  else {
+    q2_sign = 1;
+  }
+
+  // Calculate interpolation factors
+  // TODO: do we really want an epsilon so small?
+  double r, s;
+  if (std::abs(1.0 - lambda) < std::numeric_limits<double>::epsilon()) {
```

Comment:
I literally translated this from Matlab but I don't think it's right to use an epsilon this small. What do you guys think?